kits/multisig/test/multisig.js

```diff
@@ -76,270 +84,309 @@ contract('Multisig Kit', accounts => {
-            assert.equal((await voting.minAcceptQuorumPct()).toString(), multisigSupport.toString())
-            const maxUint64 = new web3.BigNumber(2).pow(64).minus(1)
-            // TODO assert.equal((await voting.voteTime()).toString(), maxUint64.toString())
-            // check that it's initialized and can not be initialized again
-            try {
-                await voting.initialize(tokenAddress, 1e18, 1e18, 1000)
-            } catch (err) {
-                assert.equal(err.receipt.status, 0, "It should have thrown")
-                return
-            }
-            assert.isFalse(true, "It should have thrown")
-        })
-
-        it('has correct permissions', async () =>{
-            const dao = await getContract('Kernel').at(daoAddress)
-            const acl = await getContract('ACL').at(await dao.acl())
-
-            const checkRole = async (appAddress, permission, managerAddress, appName='', roleName='', granteeAddress=managerAddress) => {
-                assert.equal(await acl.getPermissionManager(appAddress, permission), managerAddress, `${appName} ${roleName} Manager should match`)
-                assert.isTrue(await acl.hasPermission(granteeAddress, appAddress, permission), `Grantee should have ${appName} role ${roleName}`)
-            }
-
-            // app manager role
-            await checkRole(daoAddress, await dao.APP_MANAGER_ROLE(), votingAddress, 'Kernel', 'APP_MANAGER')
-
-            // create permissions role
-            await checkRole(acl.address, await acl.CREATE_PERMISSIONS_ROLE(), votingAddress, 'ACL', 'CREATE_PERMISSION')
-
-            // evm script registry
-            const regConstants = await getContract('EVMScriptRegistryConstants').new()
-            const reg = await getContract('EVMScriptRegistry').at(await acl.getEVMScriptRegistry())
-            await checkRole(reg.address, await reg.REGISTRY_ADD_EXECUTOR_ROLE(), votingAddress, 'EVMScriptRegistry', 'ADD_EXECUTOR')
-            await checkRole(reg.address, await reg.REGISTRY_MANAGER_ROLE(), votingAddress, 'EVMScriptRegistry', 'REGISTRY_MANAGER')
-
-            // voting
-            await checkRole(votingAddress, await voting.CREATE_VOTES_ROLE(), votingAddress, 'Voting', 'CREATE_VOTES', tokenManagerAddress)
-            await checkRole(votingAddress, await voting.MODIFY_QUORUM_ROLE(), votingAddress, 'Voting', 'MODIFY_QUORUM')
-            await checkRole(votingAddress, await voting.MODIFY_SUPPORT_ROLE(), votingAddress, 'Voting', 'MODIFY_SUPPORT')
+    // Test when organization is created in one call with `newTokenAndInstance()` and in
+    // two calls with `newToken()` and `newInstance()`
+    const creationStyles = ['single', 'separate']
+    for (const creationStyle of creationStyles) {
+        context(`> Creation through ${creationStyle} transaction`, () => {
+            let aragonId, tokenName, tokenSymbol
+
+            before(async () => {
+                aragonId = 'multisigdao-' + Math.floor(Math.random() * 1000)
```

Comment:
You would not believe how long it took me to realize that `namehash('MultisigDao-<number>.aragonid.eth')` doesn't create the same hash as the one the contract makes (`sha3(namehash('aragonid.eth') + sha3('MultisigDao-<number>'))`), due to the [UTS46 normalization](https://github.com/danfinlay/eth-ens-namehash/blob/master/index.js#L26) 😄 .

This bit was previously also giving aragonID incorrect ENS nodes to register (without the `Math.floor()`, `Math.random() * 1000` gives something like `123.1234`).